src/sqlfluff/dialects/dialect_snowflake.py

```diff
@@ -4995,6 +4995,14 @@ class CreateUserSegment(BaseSegment):
                 Ref("EqualsSegment"),
                 Ref("ObjectReferenceSegment"),
             ),
+            Sequence(
+                "TYPE",
+                Ref("EqualsSegment"),
+                OneOf(
+                    Ref("ObjectReferenceSegment"),
```

Comment:
Based on the docs, type is `PERSON | SERVICE | LEGACY_SERVICE | NULL` - i don't believe it can be quoted? I think it should be one of those 4 as keywords, not object/quoted literal

Reply:
I thought the same, but I tried with quotes and it works as well. Uppercase or lowercase too (all 4 possibilities).